Neuron/Sections/Transaction/TransactionHistoryPresenter.swift

```diff
@@ -0,0 +1,147 @@
+        let eDate: Date
+        if self.hasMoreData {
+            eDate = list.last?.date ?? Date.distantPast
+        } else {
+            eDate = Date.distantPast
+        }
+        let sentTransactions = self.sentTransactions
+        let sentList = sentTransactions.filter({ (sentTransaction) -> Bool in
+            // merge status
+            if let transaction = list.first(where: { $0.hash == sentTransaction.hash }) {
+                self.sentTransactions.removeAll(where: { $0.hash == sentTransaction.hash })
+                transaction.status = sentTransaction.status
+                return false
+            }
+            if sentTransaction.date < sDate && sentTransaction.date > eDate {
+                self.sentTransactions.removeAll(where: { $0.hash == sentTransaction.hash })
+                return true
+            } else {
+                return false
+            }
+        })
+        return (list + sentList).sorted(by: { $0.date > $1.date })
+    }
+
+    // MARK: - Load transactions
+    private var loading = false
+    private var page: UInt = 1
+    private var pageSize: UInt = 10
+    private let walletAddress: String
+    private let tokenAddress: String
+    private let tokenType: TokenModel.TokenType
+
+    private func loadData() throws -> [TransactionDetails] {
+        switch tokenType {
+        case .nervos:
+            return try AppChainNetwork().getTransactionHistory(walletAddress: walletAddress, page: page, pageSize: pageSize)
+        case .ethereum:
+            return try EthereumNetwork().getTransactionHistory(walletAddress: walletAddress, page: page, pageSize: pageSize)
+        case .erc20:
+            return try EthereumNetwork().getErc20TransactionHistory(walletAddress: walletAddress, tokenAddress: tokenAddress, page: page, pageSize: pageSize)
+        case .nervosErc20:
+            return try AppChainNetwork().getErc20TransactionHistory(walletAddress: walletAddress, tokenAddress: tokenAddress, page: page, pageSize: pageSize)
+        }
+    }
+
+    // MARK: - TransactionStatusManagerDelegate
+    func sentTransactionInserted(transaction: TransactionDetails) {
+        guard transaction.from == walletAddress else {
+            return
+        }
+        DispatchQueue.main.async {
+            self.transactions.insert(transaction, at: 0)
+            self.delegate?.didLoadTransactions(transaction: self.transactions, insertions: [0], error: nil)
+        }
+    }
+    func sentTransactionStatusChanged(transaction: TransactionDetails) {
+        DispatchQueue.main.async {
+            for (idx, trans) in self.transactions.enumerated() {
+                if trans.hash == transaction.hash {
+                    self.transactions.remove(at: idx)
+                    self.transactions.insert(transaction, at: idx)
+                    self.delegate?.updateTransactions(transaction: self.transactions, updates: [idx], error: nil)
+                    return
```

Comment:
Instead of enumerating the whole array, it's better to use `firstIndex(where:)` to find the index.